### PR TITLE
refactor: remove `Validation` from `resolvePattern` in `Resolver`

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to https://github.com/flix/flix/issues/8812

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Resolver.scala [hunk 1]
```diff
@@ -1961,49 +1955,46 @@ object Resolver {
...
         lookupTag(qname, env, ns0, root) match {
-          case Result.Ok(c) => mapN(visit(pat)) {
-            case p => ResolvedAst.Pattern.Tag(Ast.CaseSymUse(c.sym, qname.loc), p, loc)
-          }
-          case Result.Err(e) => Validation.toSoftFailure(ResolvedAst.Pattern.Error(loc), e)
+          case Result.Ok(c) =>
+            val p = visit(pat)
+            ResolvedAst.Pattern.Tag(Ast.CaseSymUse(c.sym, qname.loc), p, loc)
+          case Result.Err(error) =>
+            sctx.errors.add(error)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:1976]: This is not duplicating errors right?

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:1976]: I don't think so... it would have been duplicated before then too

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:1976]: Also, `lookupTag` returns a restult without adding anything to `sctx` so it's up to the caller what to do with the error and that's why it's added here

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:1976]: OK: let us go for it.

**Comment by jaschdoc** [comment]
Merged

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**